File: java/adaptador/midia-player/src/main/java/edu/uea/adapter/VLCAdapter.java
package edu.uea.adapter;

// Adapter 1 - Adapts VLCPlayer to MediaPlayer
public class VLCAdapter implements MediaPlayer {
    private VLCPlayer vlcPlayer;

    public VLCAdapter() {
        this.vlcPlayer = new VLCPlayer();
    }

    @Override
    public void play(String audioType, String fileName) {
        if (audioType.equalsIgnoreCase("vlc")) {
            vlcPlayer.playVLCFile(fileName);
        }
    }
}

File: python/adaptador/adapter/VLCAdapter.py
from .MediaPlayer import MediaPlayer
from .VLCPlayer import VLCPlayer 

# Adapter Class
class VLCAdapter(MediaPlayer):
    def __init__(self):
        self.vlcPlayer = VLCPlayer()

    def play(self, audio_type, file_name):
        if audio_type.lower() == "vlc":
            self.vlcPlayer.playVLCFile(file_name)